auth/range_perm_cache.go

```diff
@@ -161,6 +201,43 @@ func checkKeyPerm(cachedPerms *unifiedRangePermissions, key, rangeEnd []byte, pe
 	return false
 }
 
+// expandPrefixToRange converts a single key
+// with wildcard as its last character into a range.
+// For example, we will convert a range like ['/home/user/*', '')
+// into ['/home/user/', '/home/user0').
+// Returns true if conversion/expansion took place.
+// If removeWild is true, then we will remove the wildcard byte
+// before creating the range.
+func expandPrefixToRange(r *rangePerm, wildcard byte, removeWild bool) bool {
```

Comment:
so how would I declare a role on the prefix `abc*`? won't this turn it into a role on the prefix `abc`, matching more than intended?